### DISCUSSION TITLE
Regarding OpenAi/Swarm multi-agent collaboration

### DISCUSSION THREAD

**Discussion by AAEE86** [langgenius/dify#9717] [General]
### Self Checks

- [X] I have searched for existing issues [search for existing issues](https://github.com/langgenius/dify/issues), including closed ones.
- [X] I confirm that I am using English to submit this report (我已阅读并同意 [Language Policy](https://github.com/langgenius/dify/issues/1542)).
- [X] [FOR CHINESE USERS] 请务必使用英文提交 Issue，否则会被关闭。谢谢！:）
- [X] Please do not modify this template :) and fill in all the required fields.

### Content

Recently, OpenAi open sourced the Swarm project. I took a look at it. I feel it is very powerful.

At present, several projects I know, such as Dify, Fastgpt, and Maxkb, cannot achieve similar multi-agent collaboration.

Does Dify have any consideration in this regard?

**Comment by nickjfrench** [comment]
I'd love to hear if this is a planned feature. The concept is quite simple and implementation would probably require just allowing Agents to be built as a tool. My project end state relies heavily on this concept and I'd be disappointed if this feature was not implemented.

**Reply by seoninja13** [reply]
what is your take on it 2 month later?

**Reply by nickjfrench** [reply]
I'm not the original poster. But I'll still share my thoughts.

I'd love to have this feature implemented. I think if done correctly, you'll lower the barrier for entry for such a game changing concept (AI Swarm).

As previously mentioned, my end state solution uses this concept heavily and if Dify doesn't have the feature by then, I'll either have to move to full-code solution or heavily utilise the Dify APIs with a custom engine wrapped around

**Reply by seoninja13** [reply]
I have dabbled in Dify. Can you fully customize Dify, how extensible it is? Can you see the code base behind the UI? basically if yes, you can use UI to for your  sceleten build, and then fune tune it.. What yout think of such flow, would be worth it? I deciding the Pros and Cons of using 1 vs the other

**Comment by saendu** [comment]
Would also be interested in being able to use agents as tools in order to build multi-agents architectures (MAS). I think the only thing to think through is error and loop handlings to avoid infinite loops.
